controllers/infra/drivers/aws/stop_instance.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2022 labring.
+Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+SPDX - License - Identifier: Apache - 2.0
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package aws
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	v1 "github.com/labring/sealos/controllers/infra/api/v1"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	"github.com/aws/smithy-go"
+)
+
+// EC2StopInstancesAPI defines the interface for the StopInstances function.
+// We use this interface to test the function using a mocked service.
+type EC2StopInstancesAPI interface {
+	StopInstances(ctx context.Context,
+		params *ec2.StopInstancesInput,
+		optFns ...func(*ec2.Options)) (*ec2.StopInstancesOutput, error)
+}
+
+// StopInstance stops an Amazon Elastic Compute Cloud (Amazon EC2) instance.
+// Inputs:
+//     c is the context of the method call, which includes the AWS Region.
+//     api is the interface that defines the method call.
+//     input defines the input arguments to the service call.
+// Output:
+//     If success, a StopInstancesOutput object containing the result of the service call and nil.
+//     Otherwise, nil and an error from the call to StopInstances.
+func StopInstance(c context.Context, api EC2StopInstancesAPI, input *ec2.StopInstancesInput) (*ec2.StopInstancesOutput, error) {
+	resp, err := api.StopInstances(c, input)
+
+	var apiErr smithy.APIError
+	if errors.As(err, &apiErr) && apiErr.ErrorCode() == "DryRunOperation" {
+		fmt.Println("User has permission to stop instances.")
+		input.DryRun = aws.Bool(false)
+		return api.StopInstances(c, input)
+	}
+
+	return resp, err
+}
+
+func (d Driver) deleteInstances(hosts *v1.Hosts) error {
```

Comment:
I think we'd better add some comments here because `delete` instances are easy to obfuscate with `stop` instances here.

Reply:
aws will auto delete the stoped instance, but for this interface we want to delete the instance not stop it.